test/mini-graphql-http-client.js

```diff
@@ -93,4 +93,116 @@ describe("MiniGraphqlHttpClient tests", () => {
+                { ok: false, status: 500, statusText: "error", json: () => ({ bla: 1 }) },
+                { ok: false, status: 500, statusText: "error", json: () => ({ bla: 1 }) },
+                { ok: true, status: 200, json: () => ({ bla: 1 }) },
+                { ok: false, status: 400, statusText: "error", json: () => ({ bla: 1 }) },
+                { ok: true, status: 200, json: () => ({ bla: 1 }) },
+            ];
+            fakeFetch.calls += 1;
+            return responseArray[fakeFetch.calls];
+        }
+        fakeFetch.calls = -1;
+        return fakeFetch;
+    };
+
+    describe("#Retry", () => {
+        it("should make 6 retries on 5xx error response, but stops on 5 try, because it's get good response", async function () {
+            let wasError = false;
+            let numberOfRetries = 6;
+            const cache = {};
+            const fakeFetch = createBadAndGoodFakeFetch();
+            const client = MiniGraphqlHttpClient({
+                uri: "https://a.aa",
+                fetch: fakeFetch,
+                cache,
+                retry: numberOfRetries,
+            });
+
+            await client.query({ query: "{ bla }" }).catch(() => {
+                wasError = true;
+            });
+
+            expect(fakeFetch.calls).to.be.equal(4);
+            expect(wasError).to.be.false;
+        });
+
+        it("should make 3 retries on 5xx error response and return error", async function () {
+            let wasError = false;
+            let numberOfRetries = 3;
+            const cache = {};
+            const fakeFetch = createBadAndGoodFakeFetch();
+            const client = MiniGraphqlHttpClient({
+                uri: "https://a.aa",
+                fetch: fakeFetch,
+                cache,
+                retry: numberOfRetries,
+            });
+
+            await client.query({ query: "{ bla }" }).catch(() => {
+                wasError = true;
+            });
+
+            expect(fakeFetch.calls).to.be.equal(3);
+            expect(wasError).to.be.true;
+        });
+
+        it("should not make retries on 4xx error response and return error after 1st try", async function () {
+            let errorText;
+            let wasError = false;
+            let numberOfRetries = 3;
+            const cache = {};
+            const fakeFetch = createBadAndGoodFakeFetch();
+            fakeFetch.calls = 4; // next call will be on 400 error code
+            const client = MiniGraphqlHttpClient({
+                uri: "https://a.aa",
+                fetch: fakeFetch,
+                cache,
+                retry: numberOfRetries,
+            });
+
+            await client.query({ query: "{ bla }" }).catch((error) => {
+                errorText = error.message;
+                wasError = true;
+            });
+
+            expect(errorText).to.be.equal("400 error");
+            expect(wasError).to.be.true;
+            expect(fakeFetch.calls).to.be.equal(5);
```

Comment:
The test name is "should ... return error after 1st try"

But the code requires five.

Please, fix something here.

Reply:
on 162 line we set 4 calls so it's one call, it's can be fixed by not reusing createBadAndGoodFakeFetch